app/src/main/java/tv/vizbee/demo/vizbee/VizbeeHomeSSOAdapter.kt

```diff
@@ -0,0 +1,131 @@
+     */
+    override fun getSignInInfo(callback: ICommandCallback<List<VizbeeSignInInfo>>) {
+
+        // Implement this method to verify whether the user has signed in via
+        // the given sign-in method or not and invoke callback.onSuccess(isSignedIn)
+
+        val sharedPreferenceHelper = VizbeeWrapper.context?.get()
+            ?.let { SharedPreferenceHelper(it) }
+
+        val authToken = sharedPreferenceHelper?.getAuthToken()
+        val isSignedIn = (authToken?.isNotEmpty() == true)
+        val mobileSignInInfo =
+            VizbeeSignInInfo("MVPD", isSignedIn, JSONObject())
+        val signInInfo: List<VizbeeSignInInfo> = listOf(mobileSignInInfo)
+        callback.onSuccess(signInInfo)
+    }
+
+    /**
+     * This method authenticates the TV using the mobile auth token and the TV reg code.
+     * Use case 1: If the user has already signed in on the mobile: it invokes the backend API and
+     * passes the reg code along with the mobile auth token to authenticate the TV.
+     * Use case 2: If the user hasn't signed in on the mobile: it takes the user to the sign in page
+     * and once the sign in is done, it invokes the backend API to authenticate the TV.
+     * @param status sign in status of the TV
+     */
+    override fun onReceiveTVSignInStatus(status: VizbeeTVSignInStatus) {
+        Logger.i(
+            LOG_TAG,
+            "Received sign in status = $status context = ${VizbeeWrapper.context?.get()}"
+        )
+        if (status.vizbeeSignInState == VizbeeSignInState.SIGN_IN_IN_PROGRESS && status.signInType == "MVPD"
+        ) {
+
+            val regCode = status.customData.optString("regcode")
+            if (regCode.isNotEmpty()) {
+
+                val sharedPreferenceHelper = VizbeeWrapper.context?.get()
+                    ?.let { SharedPreferenceHelper(it) }
+
+                sharedPreferenceHelper?.saveRegCode(regCode)
+
+                if (sharedPreferenceHelper?.getAuthToken()?.isEmpty() == true) {
+                    launchLoginScreen()
+                } else {
+                    updateRegCodeStatus(sharedPreferenceHelper?.getAuthToken() ?: "")
+                }
+            } else {
+                Logger.i(LOG_TAG, "onTVSignInStatus received RegCode is empty")
+            }
+        }
+
+    }
+
+    private fun launchLoginScreen() {
+
+        val intent = Intent(VizbeeWrapper.context?.get(), MainActivity::class.java)
+        intent.putExtra(INTENT_LAUNCH_LOGIN_SCREEN, true)
+        intent.flags = Intent.FLAG_ACTIVITY_SINGLE_TOP
+        AppStateMonitor.getInstance().visibleActivity?.startActivity(intent)
+    }
+
+    fun updateRegCodeStatus(authToken: String) {
+        val apiInterface = RetrofitInstance.getInstance().create(ApiInterface::class.java)
+        val regCodeStatusRequest = RegCodeStatusRequest(
+            IDUtils.getMyDeviceID()
+        )
+        val sharedPreferenceHelper = VizbeeWrapper.context?.get()
+            ?.let { SharedPreferenceHelper(it) }
```

Comment:
If we make the SharedPreferenceHelper as singleton, we don't have to look for the context everytime

Reply:
Done